src/WebJobs.Extensions.DurableTask/ContextImplementations/RemoteEntityContext.cs

```diff
@@ -20,9 +17,41 @@ public RemoteEntityContext(EntityBatchRequest batchRequest)
         }
 
         [JsonProperty("request")]
-        public EntityBatchRequest Request { get; private set; }
+        internal EntityBatchRequest Request { get; private set; }
 
         [JsonIgnore]
         internal EntityBatchResult? Result { get; set; }
+
+        internal void EnsureSuccess()
+        {
+            if (this.Result == null)
+            {
+                throw new InvalidOperationException("Entity batch request has not been processed yet.");
+            }
+
+            if (this.Result.FailureDetails is { } f)
+            {
+                // TODO: use an entity specific exception type.
+                throw new OrchestrationFailureException(f.ErrorMessage);
+            }
+
+            List<Exception>? errors = null;
+            if (this.Result.Results is not null)
+            {
+                foreach (OperationResult result in this.Result.Results)
+                {
+                    if (result.FailureDetails is { } failure)
+                    {
+                        errors ??= new List<Exception>();
+                        errors.Add(new OrchestrationFailureException(failure.ErrorMessage));
```

Comment:
Need to create a new exception type, doesn't appear to be one in DurableTask.Core to reuse